### PR TITLE
Add "Composer Lint" job to CI pipeline

### PR DESCRIPTION
**Pull request by phansys**
Applies `composer-lint` workflow from [doctrine/.github](https://github.com/doctrine/.github/blob/87f8373f3668edfe31b6aa8d3049b3fd35f90362/.github/workflows/composer-lint.yml#L2).

### CODE REVIEW

**Review by dmaicher — APPROVED** [review]

**Review by ostrolucky — CHANGES_REQUESTED** [review]
Can you rather embed this in https://github.com/doctrine/DoctrineBundle/blob/2.7.x/.github/workflows/coding-standards.yml?

**Review by ostrolucky — APPROVED** [review]

**Comment by ostrolucky** [comment]
Hmm maybe we should have this in https://github.com/doctrine/.github/blob/main/.github/workflows/coding-standards.yml so it applies for all projects, cc @greg0ire

**Review by ostrolucky — APPROVED** [review]

**Comment by greg0ire** [comment]
I think it could indeed be contributed to that repo. If we add it to the CS workflow. If we do that, the next release should probably be 2.0.0 to make it clear that an action must be taken to make the build green (fixing `composer.json`). It would be great to make this workflow kick in only when appropriate, using [`paths`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths), otherwise it will mean one more job per project, and a job that is unlikely to fail when `composer.json` is not touched.

File: .github/workflows/coding-standards.yml [hunk 1]
```diff
@@ -15,3 +15,7 @@ jobs:
     uses: "doctrine/.github/.github/workflows/coding-standards.yml@1.4.1"
     with:
       composer-options: "--prefer-dist --prefer-stable"
+
+  lint-composer:
+    name: "Lint Composer"
+    uses: "phansys/doctrine.github/.github/workflows/composer-lint.yml@composer_lint"
```
Comment (greg0ire) [inline .github/workflows/coding-standards.yml]: This needs to be in a separate workflow so that you can use `paths`

Comment (stof) [inline .github/workflows/coding-standards.yml]: This should not reference your fork

**Review by stof — DISMISSED** [review]

File: .github/workflows/coding-standards.yml (hunk 1 above)
Comment (greg0ire) [reply .github/workflows/coding-standards.yml]: It's temporary, for test purposes, in order to proove that the upstream PR works before merging it.

Comment (greg0ire) [reply .github/workflows/coding-standards.yml]: Let's convert this to a draft in the meantime.

**Review by greg0ire — APPROVED** [review]
:ok_hand:

**Review by ostrolucky — APPROVED** [review]

**Pull request merged**